src/pages/learn.js

```diff
@@ -1,41 +1,50 @@
-import React, { useContext, useEffect } from "react"
+import React, { useContext, useEffect, useState } from "react"
 import Navbar from "../components/navbar"
 import IDE from "../components/ide"
 import Term from "../components/term"
 import Video from "../components/video"
-import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
+import { HeartwoodStateContext } from "../state/HeartwoodContextProvider"
+
+import {signUpComplete} from "../utils/utils"
 
 import "../styles/video.css"
 
 const Learn = ({ location }) => {
   const state = useContext(HeartwoodStateContext)
 
-  // Jacob, make sure to turn this back to || false when you are done testing
-  const { showPage } = location.state || false
+  const [showPage, setShowPage] = useState(false)
 
   useEffect(() => {
+    if (state.user) {
     const gatsby = require("gatsby")
-    if (!showPage) {
+    console.log("location is", location.state)
+    if (true) {
+      setShowPage(true)
+    } else {
       gatsby.navigate("/")
     }
-  }, [])
+    
+    }
+  }, [state.user])
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'location.state'. Either include it or remove the dependency array.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5919912)